controllers/istiorevisiontag/istiorevisiontag_controller.go

```diff
@@ -0,0 +1,541 @@
+	c.Message = fmt.Sprintf("failed to determine if revision tag is in use: %v", err)
+	return c, fmt.Errorf("failed to determine if IstioRevisionTag is in use: %w", err)
+}
+
+func (r *Reconciler) updateLabels(ctx context.Context, tag *v1alpha1.IstioRevisionTag, rev *v1alpha1.IstioRevision) error {
+	updatedTag := tag.DeepCopy()
+	if rev == nil {
+		delete(updatedTag.Labels, sailOperatorReferencedRevisionLabel)
+	} else {
+		if updatedTag.Labels == nil {
+			updatedTag.Labels = make(map[string]string, 1)
+		}
+		updatedTag.Labels[sailOperatorReferencedRevisionLabel] = rev.Name
+	}
+	return r.Patch(ctx, updatedTag, client.MergeFrom(tag))
+}
+
+func (r *Reconciler) isRevisionTagReferencedByWorkloads(ctx context.Context, tag *v1alpha1.IstioRevisionTag) (bool, error) {
+	log := logf.FromContext(ctx)
+	nsList := corev1.NamespaceList{}
+	nsMap := map[string]corev1.Namespace{}
+	if err := r.Client.List(ctx, &nsList); err != nil { // TODO: can we optimize this by specifying a label selector
+		return false, fmt.Errorf("failed to list namespaces: %w", err)
+	}
+	for _, ns := range nsList.Items {
+		if namespaceReferencesRevisionTag(ns, tag) {
+			log.V(2).Info("RevisionTag is referenced by Namespace", "Namespace", ns.Name)
+			return true, nil
+		}
+		nsMap[ns.Name] = ns
+	}
+
+	podList := corev1.PodList{}
+	if err := r.Client.List(ctx, &podList); err != nil { // TODO: can we optimize this by specifying a label selector
+		return false, fmt.Errorf("failed to list pods: %w", err)
+	}
+	for _, pod := range podList.Items {
+		if podReferencesRevisionTag(pod, tag) {
+			log.V(2).Info("RevisionTag is referenced by Pod", "Pod", client.ObjectKeyFromObject(&pod))
+			return true, nil
+		}
+	}
+
+	rev, err := r.getIstioRevision(ctx, tag.Spec.TargetRef)
+	if err != nil {
+		return false, err
+	}
+
+	if tag.Name == v1alpha1.DefaultRevision && rev.Spec.Values != nil &&
+		rev.Spec.Values.SidecarInjectorWebhook != nil &&
+		rev.Spec.Values.SidecarInjectorWebhook.EnableNamespacesByDefault != nil &&
+		*rev.Spec.Values.SidecarInjectorWebhook.EnableNamespacesByDefault {
+		return true, nil
+	}
+
+	log.V(2).Info("RevisionTag is not referenced by any Pod or Namespace")
+	return false, nil
+}
+
+func namespaceReferencesRevisionTag(ns corev1.Namespace, tag *v1alpha1.IstioRevisionTag) bool {
+	return tag.Name == getReferencedRevisionFromNamespace(ns.Labels)
+}
+
+func podReferencesRevisionTag(pod corev1.Pod, tag *v1alpha1.IstioRevisionTag) bool {
+	return tag.Name == getReferencedRevisionTagFromPod(pod.GetLabels())
+}
+
+func getReferencedRevisionFromNamespace(labels map[string]string) string {
+	if labels[IstioInjectionLabel] == IstioInjectionEnabledValue {
+		return v1alpha1.DefaultRevision
+	}
+	revision := labels[IstioRevLabel]
+	if revision != "" {
+		return revision
+	}
+	// TODO: if .Values.sidecarInjectorWebhook.enableNamespacesByDefault is true, then all namespaces except system namespaces should use the "default" revision
```

Comment:
Do we need this TODO here as we are checking for the enableNamespacesByDefault in Line 396?